evergreen/config.yml

```diff
@@ -33,83 +33,134 @@ functions:
+          fi
 
-        set_cmake_var realm_vars REALM_BUILD_COMMANDLINE_TOOLS BOOL On
-        set_cmake_var realm_vars REALM_ENABLE_ENCRYPTION BOOL On
+          if [ -n "${use_system_openssl|}" ]; then
+              set_cmake_var realm_vars REALM_USE_SYSTEM_OPENSSL BOOL On
+          fi
 
-        if [[ -n "${fetch_missing_dependencies|}" ]]; then
-            set_cmake_var realm_vars REALM_FETCH_MISSING_DEPENDENCIES BOOL On
-        fi
+          set_cmake_var realm_vars REALM_BUILD_COMMANDLINE_TOOLS BOOL On
+          set_cmake_var realm_vars REALM_ENABLE_ENCRYPTION BOOL On
+
+          if [[ -n "${fetch_missing_dependencies|}" ]]; then
+              set_cmake_var realm_vars REALM_FETCH_MISSING_DEPENDENCIES BOOL On
+          fi
+
+          if [[ -n "${cmake_build_type|}" ]]; then
+              set_cmake_var realm_vars CMAKE_BUILD_TYPE STRING "${cmake_build_type}"
+          fi
+
+          echo "Running cmake with these vars:"
+          cat cmake_vars/*.txt | tee cmake_vars.txt
+          echo
+
+          $CMAKE \
+            -B build \
+            -C cmake_vars.txt ${extra_flags} \
+            -G "${cmake_generator|Unix Makefiles}"
+
+          ${cmake_bindir}/cmake \
+              --build build \
+              -j ${max_jobs|$(grep -c proc /proc/cpuinfo)} \
+              --target ${target_to_build|}
+
+  "run benchmark":
+    - command: shell.exec
+      params:
+        working_dir: realm-core
+        shell: bash
+        script: |-
+          set -o errexit
+
+          if [[ -z "${path_to_benchmark}" ]]; then
+              echo "No path to benchmark specified."
+              exit 1
+          fi
+
+          BENCHMARK=$(./evergreen/abspath.sh ${path_to_benchmark})
+          echo "Going to run benchmark $BENCHMARK"
+
+          [[ -d benchmark_results ]] && rm -rf benchmark_results
+          mkdir benchmark_results
+          cd benchmark_results
+
+          $BENCHMARK
+    - command: perf.send
+      params:
+        file: './realm-core/benchmark_results/results.latest.json'
+
+  "run tests":
+    - command: shell.exec
+      params:
+        working_dir: realm-core
+        shell: bash
+        script: |-
+          set -o errexit
+          CTEST=$(pwd)/${cmake_bindir}/ctest
+
+          if [[ -n "${test_filter}" ]]; then
+              TEST_FLAGS="-R ${test_filter}"
+          fi
+          TETS_FLAGS="$TEST_FLAGS ${test_flags|}"
+
+          cd build
+          $CTEST -V $TEST_FLAGS
+
+tasks:
+- name: compile
+  commands:
+  - func: "compile"
 
-        echo "Running cmake with these vars:"
-        cat cmake_vars/*.txt | tee cmake_vars.txt
-        echo
-        
-        mkdir build
-        $CMAKE \
-          -B build \
-          -C cmake_vars.txt ${extra_flags} \
-          -G "${cmake_generator|Unix Makefiles}"
-
-        ${cmake_bindir}/cmake --build build -j ${max_jobs|$(grep -c proc /proc/cpuinfo)}
 - name: package
-  depends_on:
-  - "compile"
   commands:
+  - func: "compile"
```

Comment:
Instead of depending on compile, the test tasks will just run compile at the beginning. Since they all execute serially on a single machine this means the compile should be a noop, but it also means we can restart/run the tests independently in a patch build.